src/shims/env.rs

```diff
@@ -465,4 +465,35 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         Ok(())
     }
+
+    fn getpid(&mut self) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+        let target_os = &this.tcx.sess.target.os;
+        assert!(
+            target_os == "linux" || target_os == "macos",
+            "`getpid` is only available for the UNIX target family"
+        );
+
+        this.check_no_isolation("`getpid`")?;
+
+        // The reason we need to do this wacky of a conversion is because
+        // `libc::getpid` returns an i32, however, `std::process::id()` return an u32,
+        // So we need to fit that u32 into an i32 and pray
```

Comment:
For what it's worth, both have the same size -- so we can also just return a `u32`, can't we? I assume that is what happens in std, it will convert negative PIDs to large `u32` PIDs.

So maybe in this special case, `as i32` is actually okay. I wish there was a `wrapping_from` function...

Reply:
I don't see any reason why there would be a problem here. Just return `std::process::id() as i32` and make a comment that this is a u32-to-i32 cast and we are okay with the wrapping behavior since stdlib does the same thing (in the opposite direction) internally. Of course you should first verify that this is indeed what the stdlib does. :)

We avoid `as` casts to avoid accidentally wrapping or truncating, but in this case I think after deliberation we conclude that wrapping/truncation is okay.

Reply:
Okay I will do this instead

Reply:
So, like that?